### ISSUE TITLE
help wanted!

### ISSUE THREAD

**Issue by rabernat** [xgcm/xgcm#104]
This issue is just to make it clear that I am explicitly looking for volunteers to take on responsibility for versioning (#101), packaging (#100), documentation, and overall administration of xgcm. I just don't have time to tackle these things any time soon, but they are important for the long-term sustainability of this project.

For example, @jbusecke, @raphaeldussin, @willirath, @maikejulie, @jetesdal or anyone else who wants to learn more about how that stuff works, this is an opportunity to do so...

**Comment by raphaeldussin** [comment]
count me in!

**Comment by jbusecke** [comment]
Yup I would be interested!

**Comment by rabernat** [comment]
Great, thanks! I have given you both write access, so you can commit / merge PRs. (I would strongly prefer that all changes continue to come through PRs.) I probably also have to give you permissions on pypi--I'll try to sort that out asap.

So basically just start doing it. I recommend starting by copying xarray's "how to release" instructions / adapting them for xgcm:
https://github.com/pydata/xarray/blob/master/HOW_TO_RELEASE

**Comment by rabernat** [comment]
In order to get permissions to publish on pypi, you need to make accounts there and then tell me your usernames. I will add you to the xgcm project: https://pypi.org/project/xgcm/

**Comment by willirath** [comment]
I'm happy to join and look into #100.

(But I'll be quite busy the next few days.)


Am 14. Juni 2018 02:45:07 schrieb Ryan Abernathey <notifications@github.com>:

**Comment by rabernat** [comment]
Just to be clear, the near-term goal is to release xgcm v 0.2.0 via pip and conda-forge.

**Comment by jbusecke** [comment]
Do we have concrete milestones for 0.2.0 (e.g. working metrics module and perhaps vector calculus?)

**Comment by jbusecke** [comment]
Joined PyPI (username: jbusecke)

**Comment by raphaeldussin** [comment]
same (username: raphaeldussin)

**Comment by rabernat** [comment]
Ok you are both added as maintainers.

Let's make 0.2.0 just the current state of the project. No need to wait for those hard things to get done. We have not had a release in a long time, and there are already many improvements that we need to get out there.

We can discuss milestones for 0.3.0 in a separate issue.

**Comment by raphaeldussin** [comment]
from the pypi and conda docs, it seems that the way to go is to start with creating the pypi package
then the conda recipe. I can start testing it next week if that's ok with everybody.

**Comment by maikejulie** [comment]
Happy to be involved, but a bit busy out June sorry!

**Comment by jbusecke** [comment]
Sounds great. I have started a todo list (#105), where we can discuss what is still missing for a clean release of v0.2.0. Perhaps its only the things that I have to do...then this might have been a bit overkill. 

Either way let me know if there are things (particularly in the documentation) that are still needed.

**Comment by rabernat** [comment]
> from the pypi and conda docs, it seems that the way to go is to start with creating the pypi package
> then the conda recipe. I can start testing it next week if that's ok with everybody.

Note that @ocefpaf has already created a conda-forge package for us (https://github.com/conda-forge/xgcm-feedstock), but it needs some attention (see discussion here: https://github.com/conda-forge/staged-recipes/pull/6056#issuecomment-397283231). He would be a great person to ask if you have questions about releasing.

As for the overall release procedure, I highly recommend you follow the procedure described in the xarray HOW_TO_RELEASE document:
https://github.com/pydata/xarray/blob/master/HOW_TO_RELEASE

**Comment by jbusecke** [comment]
Now that we have a more active development team should we close this? If we still want to highlight this issue, can we pin it somehow to the top? Otherwise it will not be seen I am afraid.

**Comment by jbusecke** [comment]
Ah there it is. I pinned this to the top.

**Comment by ocefpaf** [comment]
> He would be a great person to ask if you have questions about releasing.

I usually recommend people to take a look at https://github.com/ioos/ioos-python-package-skeleton

> As for the overall release procedure, I highly recommend you follow the procedure described in the xarray HOW_TO_RELEASE document:
> [pydata/xarray@`master`/HOW_TO_RELEASE](https://github.com/pydata/xarray/blob/master/HOW_TO_RELEASE)

Got a broken link in that but `xarray` is a bigger project and needs more human intervention for a release. This project is in good shape and I would recommend something simpler and automated:

- dropping versioneer, which is deprecated, and adopting setuptools_scm;
- adding some sdist tarball/wheels tests (see link above);
- use a GH action to release every new tag.

I can send PRs to implent those if you want to adopt that wokflow.

**Comment by jbusecke** [comment]
I would like to keep this open, since it is a general announcement?

**Comment by TomNicholas** [comment]
Anyone interested in the (lack of) maintainers of xGCM right now will be interested to read this thread

https://discourse.pangeo.io/t/status-of-xgcm/4986

**Comment by raphaeldussin** [comment]
it looks like I used to be in the maintainer's list but not anymore. I can try to advertise to a folks at GFDL who use the package in their workflow

**Comment by TomNicholas** [comment]
Thanks @raphaeldussin - you should have just been sent an invite that will give you permissions on this repo. You will then have the power to add anyone else.

**Comment by TomNicholas** [comment]
Amazing @NoraLoose ! You guys at [C]Worthy are perfectly placed for that. 

What I really want to do is be able to do some kind of hand-off. I mentioned on the discourse thread that I'm happy to sit down for a few hours anytime to talk people through confusing parts of this repository, but I'm not going to write/review any code.

**Comment by hdrake** [comment]
@NoraLoose and @raphaeldussin, I propose the three of us attempt to release v0.8.2 in the near future as a way of getting up to speed on the state of the repo and the package maintenance process. What do you think?

We can reach out to @TomNicholas if (likely when) we run into any roadblocks or have amassed a list of questions for him.

**Comment by hdrake** [comment]
@rabernat, @TomNicholas, @jbusecke, it seems there are (at least) three loose ends to complete the turnover of maitaining xgcm to myself (@hdrake), @NoraLoose, and @raphaeldussin.

1) Add us to the ReadTheDocs xgcm project so we can update the docs (see https://github.com/xgcm/xgcm/issues/669#issuecomment-2738118060)
2) Add us to the [xgcm-feedstock](https://github.com/conda-forge/xgcm-feedstock) repo so we can approve new releases on conda
3) Decide whether we should decouple `xgcm` documentation from Pangeo and cloud storage or somehow continue maintenance of [xgcm-examples](https://github.com/xgcm/xgcm-examples), which would require all maintainers/community developers to have access to cloud-based data.

**Comment by TomNicholas** [comment]
1. Add us to the ReadTheDocs xgcm project
2. Add us to the [xgcm-feedstock](https://github.com/conda-forge/xgcm-feedstock) repo

Unfortunately I can't help you with either of these, but I will nag Ryan / Julius to help you next week.

**Comment by NoraLoose** [comment]
> 1. Add us to the ReadTheDocs xgcm project so we can update the docs (see https://github.com/xgcm/xgcm/issues/669#issuecomment-2738118060)
> 2. Add us to the [xgcm-feedstock](https://github.com/conda-forge/xgcm-feedstock) repo so we can approve new releases on conda
> 3. Decide whether we should decouple xgcm documentation from Pangeo and cloud storage or somehow continue maintenance of [xgcm-examples](https://github.com/xgcm/xgcm-examples), which would require all maintainers/community developers to have access to cloud-based data.

4. Add us as Maintainers to the [PyPI xgcm project](https://pypi.org/project/xgcm/). :)

**Comment by hdrake** [comment]
I came across [this comment](https://github.com/conda-forge/staged-recipes/pull/6056#issuecomment-397283231) and [created a PR](https://github.com/conda-forge/xgcm-feedstock/pull/18) to add @NoraLoose and I as maintainers.

**Comment by hdrake** [comment]
Gently pinging @rabernat, @jbusecke, and @TomNicholas about the following:

> 1. Add us (@hdrake, @NoraLoose, @raphaeldussin) to the ReadTheDocs xgcm project so we can update the docs (see [[BUG] RTD build is failing #669 (comment)](https://github.com/xgcm/xgcm/issues/669#issuecomment-2738118060))
> 2. Add us (@hdrake, @NoraLoose, @raphaeldussin) to the [xgcm-feedstock](https://github.com/conda-forge/xgcm-feedstock) repo so we can approve new releases on conda
> 3. Add us (@hdrake, @NoraLoose, @raphaeldussin) as Maintainers to the [PyPI xgcm project](https://pypi.org/project/xgcm/)

Separately, we @NoraLoose and I would appreciate advice as to how to proceed on this:
> Decide whether we should decouple xgcm documentation from Pangeo and cloud storage or somehow continue maintenance of [xgcm-examples](https://github.com/xgcm/xgcm-examples), which would require all maintainers/community developers to have access to cloud-based data.

**Comment by jbusecke** [comment]
Hey folks, just updating a few things here (thanks for the patience). 

@hdrake is now co-owner of the pypi project (@NoraLoose @raphaeldussin please provide @hdrake with your pypi usernames)

@rabernat you are the single owner/maintainer of the RTD project, could you add all of us to that one (this is the only thing that is needed from your end).

**Comment by jbusecke** [comment]
Also made @NoraLoose and @hdrake co-admins of xgcm-examples for now. I believe how to refactor the docs will require some more careful discussion, but just to remove bottlenecks for now.

**Comment by hdrake** [comment]
Thanks @jbusecke–I think we are just waiting on @rabernat to add the following usernames [as maintainers on RTD](https://docs.readthedocs.com/platform/stable/guides/managing-maintainers.html#adding-a-maintainer):
* `hdrake`
* `NoraLoose`
* `raphaeldussin`

**Comment by rabernat** [comment]
I invited `hdrake` and `raphaeldussin`. The user `NoraLoose` doesn't exist on RTD, but anyway, now you should be able to add other maintainers by youself.

Thanks so much for taking this on! 💪

**Comment by jbusecke** [comment]
Thanks @rabernat! 

@hdrake I could make some time to triage the release next week if that would work with y'all.

**Comment by hdrake** [comment]
I think we (@NoraLoose, @raphaeldussin, @hdrake, with help from @VeckoTheGecko and @jbusecke) have successfully taken over maintaining `xgcm`.

A few months ago we put out the first release in a while, v0.9.0. We were able to propagate this through PyPI and conda-forge. However, the one thing remaining is that the Zenodo webhook is broken, presumably because @rabernat's access token expired.

@rabernat, how would you like to proceed?
(1) transfer the ownership of the [xgcm Zenodo record](https://doi.org/10.5281/zenodo.826925) to me, allowing us to keep the same concept DOI going
(2) I start a new Zenodo record (with new concept DOI) for versions ≥0.9.0
